### DISCUSSION TITLE
How to compile?

### DISCUSSION THREAD

**Discussion by NonaSuomy** [pschatzmann/arduino-audio-tools#1504] [Q&A] [answered]
Hello, what am I missing to get this to compile?

In file included from c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/AudioHttp.h:8,
                 from c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioTools.h:94,
                 from C:\Users\Support\AppData\Local\Temp\.arduinoIDE-unsaved2024313-6068-vkqoir.a4t4g\sketch_apr13a\sketch_apr13a.ino:10:
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:272:3: error: 'WiFiClient' does not name a type
  272 |   WiFiClient* clientInsecure = nullptr; // wifi client for http
      |   ^~~~~~~~~~
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:272:3: note: the macro 'WiFiClient' had not yet been defined
In file included from C:\Users\Support\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.0-rc1\libraries\WiFi\src/WiFi.h:38,
                 from c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/AudioServer.h:10,
                 from c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/AudioHttp.h:10:
C:\Users\Support\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.0-rc1\libraries\WiFi\src/WiFiClient.h:3: note: it was later defined here
    3 | #define WiFiClient NetworkClient
      | 
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h: In destructor 'virtual audio_tools::URLStream::~URLStream()':
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:57:9: error: 'clientInsecure' was not declared in this scope; did you mean 'clientSecure'?
   57 |     if (clientInsecure != nullptr) {
      |         ^~~~~~~~~~~~~~
      |         clientSecure
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:58:14: error: type '<type error>' argument given to 'delete', expected pointer
   58 |       delete clientInsecure;
      |              ^~~~~~~~~~~~~~
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h: In member function 'Client& audio_tools::URLStream::getClient(bool)':
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:371:9: error: 'clientInsecure' was not declared in this scope; did you mean 'clientSecure'?
  371 |     if (clientInsecure == nullptr) {
      |         ^~~~~~~~~~~~~~
      |         clientSecure
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:372:28: error: expected type-specifier before 'WiFiClient'
  372 |       clientInsecure = new WiFiClient();
      |                            ^~~~~~~~~~
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:375:13: error: 'clientInsecure' was not declared in this scope; did you mean 'clientSecure'?
  375 |     return *clientInsecure;
      |             ^~~~~~~~~~~~~~
      |             clientSecure
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h: In member function 'bool audio_tools::URLStream::login()':
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:398:9: error: 'WiFi' was not declared in this scope
  398 |         WiFi.status() != WL_CONNECTED) {
      |         ^~~~
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:398:26: error: 'WL_CONNECTED' was not declared in this scope
  398 |         WiFi.status() != WL_CONNECTED) {
      |                          ^~~~~~~~~~~~
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:409:12: error: 'WiFi' was not declared in this scope
  409 |     return WiFi.status() == WL_CONNECTED;
      |            ^~~~
c:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools\src/AudioHttp/URLStream.h:409:29: error: 'WL_CONNECTED' was not declared in this scope
  409 |     return WiFi.status() == WL_CONNECTED;
      |                             ^~~~~~~~~~~~

Using library audio-tools at version 0.9.6 in folder: C:\Users\Support\Documents\Arduino\libraries\arduino-audio-tools 
Using library NetworkClientSecure at version 2.0.0 in folder: C:\Users\Support\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.0-rc1\libraries\NetworkClientSecure 
Using library Networking at version 1.0.0 in folder: C:\Users\Support\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.0-rc1\libraries\Network 
Using library WiFi at version 2.0.0 in folder: C:\Users\Support\AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.0-rc1\libraries\WiFi 
exit status 1

Compilation error: exit status 1

**Comment by pschatzmann** [comment, marked as answer]
URLStream is part of the core functionality: just include "AudioTools.h" like in [this example](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-stream/streams-url_mp3_helix-i2s/streams-url_mp3_helix-i2s.ino).

It will work when your processor supports it! So what Micrcontroller are you using ?

**Reply by pschatzmann** [reply]
This seems to be a feature of 3.0.0-rc1: I committed a correction!

**Reply by NonaSuomy** [reply]
ESP32-S3-N16R8 Note I switched to 2.0.15 and it compiled after I realized I was using dev.
